core/services/gateway/handlers/handler.go

```diff
@@ -29,7 +32,7 @@ type Handler interface {
 	// Each user request is processed by a separate goroutine, which:
 	//   1. calls HandleUserMessage
 	//   2. waits on callbackCh with a timeout
-	HandleUserMessage(ctx context.Context, msg *api.Message, callbackCh chan<- UserCallbackPayload) error
+	HandleUserMessage(ctx context.Context, jsonRequest jsonrpc2.Request, msg *api.Message, callbackCh chan<- UserCallbackPayload) error
```

Comment:
```suggestion
	HandleUserMessage(ctx context.Context, msg *api.Message, callbackCh chan<- UserCallbackPayload) error
		HandleNewUserMessage(ctx context.Context, jsonRequest jsonrpc2.Request, callbackCh chan<- UserCallbackPayload) error
```

and implement a noop HandleNewUserMessage in existing services